packages/blade/src/components/Radio/useRadio.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import React from 'react';
+import type { GestureResponderEvent } from 'react-native';
+import { useControllableState } from '../../hooks/useControllable';
+import { useId } from '../../hooks/useId';
+import { getPlatformType, makeAccessible } from '../../utils';
+import type { RadioProps } from './Radio';
+
+type UseRadioProps = Pick<
+  RadioProps,
+  'isChecked' | 'defaultChecked' | 'isDisabled' | 'isRequired' | 'onChange' | 'name' | 'value'
+> & { hasError?: boolean };
+
+const useRadio = ({
+  isChecked,
+  defaultChecked,
+  isDisabled,
+  isRequired,
+  hasError,
+  onChange,
+  name,
+  value,
+}: UseRadioProps) => {
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const isReactNative = getPlatformType() === 'react-native';
+  if (isChecked && defaultChecked) {
+    throw new Error(
+      `[Blade useRadio] Do not provide both 'isChecked' and 'defaultChecked' to useRadio. Consider if you want this component to be controlled or uncontrolled.`,
```

Comment:
If I as a consumer pass `isChecked` and `defaultChecked` will I get the same error? Seems misleading for a consumer since it talks about `useRadio`. If this is consumer facing then we should rephrase this to make it more generic for Radio

Reply:
Sure.